packages/core/src/utils/viaProxy.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Utility to redirect remote sources via Proxy
+ *
+ * This utility is used to rewrite a request to be routed through
+ * the Proxy instead.
+ */
+
+/**
+ * Virtual Module provided by the Motion Canvas
+ * Plugin. Returns `false` if proxy is set to `false`, else `true`
+ */
+//@ts-ignore
+import {enabled} from 'virtual:proxysettings';
+
+/**
+ * Helper-Function to route remote requests through a local proxy.
+ *
+ * @example This rewrites a remote url like `https://via.placeholder.com/300.png/09f/fff` into a URI-Component-Encoded string like `/cors-proxy/https%3A%2F%2Fvia.placeholder.com%2F300.png%2F09f%2Ffff`
+ */
+export function viaProxy(url: string, overrideProxySetting?: boolean) {
```

Comment:
This method should probably be called `useProxy()` to fit in with the naming convention for the other utility functions.

Reply:
My main motivation behind not using the `use`-Prefix is that, looking at the other utils, `use*` is used for getting some sort of dependency.

`useRandom` returns a Random-Instance, `useThread` returns a reference to the thread, etc.
`useProxy` would, in my opinion, mean that you would get some sort of Proxy-Instance.

Reply:
`useProxy` is kind of returning a dependency, since it's transforming an image URL into a project-dependant and usable URL. Also, this project is heavily based on the architecture of React. In React, hook function start with `use` to signify that they aren't pure, and that you're depending on a functionality of React. In this case, you'd be depending on the proxy functionality, so I'd argue the `use` naming scheme would be appropriate.

Reply:
I agree with @WaldemarLehner. The `use` prefix is used for hooks - things that can only be called in the context of the animation. Calling `useScene` outside of the animation (for example in a promise) will fail. The same goes for other hooks. `viaProxy` is just a simple wrapper function that can be called anywhere.